.github/workflows/publish_docs.yaml

```diff
@@ -0,0 +1,54 @@
+name: Publish docs
+
+on:
+  push:
+    branches:
+      - main
+      - v*.*
+
+env:
+  ELIXIR_OPTS: "--warnings-as-errors"
+  ERLC_OPTS: "warnings_as_errors"
+  LANG: C.UTF-8
+  AWS_S3_BUCKET: "s3.hex.pm"
+
+jobs:
+  release_pre_built:
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - otp: 24
+            otp_version: 24.3
+          - otp: 25
+            otp_version: 25.0
+            build_docs: build_docs
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 50
+      - uses: ./.github/workflows/release_pre_built
+        with:
+          otp_version: ${{ matrix.otp_version }}
+          otp: ${{ matrix.otp }}
+          build_docs: ${{ matrix.build_docs }}
+      - uses: gilacost/branch-tag-action@v1.3
+        id: extract
+      - name: Upload Docs to S3
+        if: ${{ matrix.build_docs }}
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          version=$(echo ${{ env.BRANCH_TAG }} | sed -e 's/^v//g' | sed -e 's/\//-/g')
+          for f in doc/*; do
+              if [ -d "$f" ]; then
+                  app=`echo $f | sed s/"doc\/"//`
+                  tar -czf "${app}-${version}.tar.gz" -C "doc/${app}" .
+                  aws s3 cp "${app}-${version}.tar.gz" "s3://${{ env.AWS_S3_BUCKET }}/docs/${app}-${version}.tar.gz" \
+                      --cache-control "public,max-age=3600" \
+                      --metadata "{\"surrogate-key\":\"docs/${app}-${version}\",\"surrogate-control\":\"public,max-age=604800\"}"
```

Comment:
We will also need to purge the fastly cache key here https://github.com/hexpm/bob/blob/ab6e9db2df0ed0abcf1b5c81178d2c83237b267c/priv/scripts/utils.sh#L3.

Reply:
I just updated the action with the `fastly_purge` [curl](https://github.com/elixir-lang/elixir/pull/12235/files#diff-4de995e1609d8ad989b4542c46a7789c934599bab17bab700944864a8acaee27R51), please let me know if this will be enought as I am not able to test this in any way. This will riquire to have the `BOB_FASTLY_KEY` secret in place.